### DISCUSSION TITLE
What is Avalonia.Skia For?

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#7629] [Q&A] [answered]
I see a new repository https://github.com/AvaloniaUI/Avalonia.Skia.

I'm curious what this is for?

I know SkiaSharp has been poorly maintained lately. Is this an attempt for Avalonia to provide it's own C# bindings for Skia itself to cut out SkiaSharp and keep things current with upstream Skia?

**Comment by jmacato** [comment, marked as answer]
@robloo yes, you're exactly right :)

**Reply by robloo** [reply]
Very interesting. I'll keep a close eye on this then!

**Reply by ahopper** [reply]
I wonder if this https://github.com/dotnet/runtime/issues/60595 might be useful for this.

**Reply by kekekeks** [reply]
There will be only 1 (one) `DllImport`-ed function.

**Reply by ahopper** [reply]
Ahh the old DoWhateverMyMagicStructSays(MagicStruct) ploy :)

**Reply by kekekeks** [reply]
Nope, it will be exactly the same as Avalonia.Native.

**Reply by ahopper** [reply]
cool, looking forward to seeing it, hopefully it gives opportunities for things like reducing allocations for changing text.

**Reply by bmitc** [reply]
@jmacato @kekekeks Will Avalonia.Skia be usable outside of Avalonia as a replacement for SkiaSharp? I.e., will it simply be a .NET 6+ library with Skia bindings *and* binaries packaged along? I want to use Skia to draw to a GLFW window, backed by either OpenGL or Vulkan, using F#.

**Reply by Costigan** [reply]
I have the same question as @bmitc.  Will Avolonia.Skia be usable outside of Avalonia as a replacement for SkiaSharp (but built on a more current skia milestone)?

**Reply by maxkatz6** [reply]
> Will Avolonia.Skia be usable outside of Avalonia as a replacement for SkiaSharp (but built on a more current skia milestone)?

No. It was planned as a limited subset of skia, that is needed for avalonia to work. No image processing, no image metadata, no image decoding... At least until Avalonia itself will need it.
SkiaSharp is a better choice for multipurpose image processing anyway.

**Reply by Costigan** [reply]
Thank you for a very quick and clear reply.  It's unfortunate that SkiaSharp's development seems to have slowed way down and gotten so far behind Skia.  I wish it were clearer to the community what's going on and whether SkiaSharp will ever support bug fixes and features in Skia milestones later than m88.

**Comment by robloo** [comment]
@jmacato I see the repo is now private. Any reason we can't follow along?

**Reply by bmitc** [reply]
I would also be interested in following along if not using bits of it. I am looking for SkiaSharp alternatives.

**Reply by Costigan** [reply]
I'd be interested too.  I'm also looking for a SkiaSharp alternative.  My use case doesn't need image metadata.  I can roll my own image decoding using another library.  I'm not sure what's meant by image processing.  I need font management, vector and raster drawing operations, and management of the OpenGL surface.  The thing that's driving me to look for SkiaSharp alternatives is a need for the somewhat more general shader support that later Skia milestones implement.
